benchopt/base.py

```diff
@@ -517,3 +518,45 @@ def __reduce__(self):
             self._module_filename, module_hash, self._parameters, dataset,
             str(self._import_ctx._benchmark_dir)
         )
+
+    def default_split(self, *arrays):
+        i_train, i_test = next(self._cv)
+        res = ()
+        for x in arrays:
+            res = (*res, x[i_train], x[i_test])
+        return res
+
+    def custom_split(self, *arrays):
+        if not hasattr(self, "_cv"):
+            # this define _cv only once
+            self._cv = itertools.cycle(self.cv.split(*arrays))
+        i_train, i_test = next(self._cv)
+        return self.split(i_train, i_test)
```

Comment:
It may be that `cv` returns something more complex, so I would be more generic:
```suggestion
        split_data = next(self._cv)
        return self.split(*split_data)
```

Reply:
But if we say to the user that he needs to define a cv object from sklearn, next(cv) will always return train_index, test_index no?